test/Microsoft.NET.TestFramework/TestAsset.cs

```diff
@@ -128,21 +131,28 @@ public TestAsset ReplacePackageVersionVariable(string targetName, string targetV
 
             return WithProjectChanges(project =>
             {
-                var ns = project.Root.Name.Namespace;
-                foreach (var PropertyName in PropertyNames)
+                if (project.Root is not null)
                 {
-                    var packageReferencesToUpdate =
-                        project.Root.Descendants(ns + PropertyName)
-                            .Where(p => p.Attribute("Version") != null && p.Attribute("Version").Value.Equals($"$({targetName})", StringComparison.OrdinalIgnoreCase));
-                    foreach (var packageReference in packageReferencesToUpdate)
+                    var ns = project.Root.Name.Namespace;
+                    foreach (var PropertyName in PropertyNames)
                     {
-                        packageReference.Attribute("Version").Value = targetValue;
+                        var packageReferencesToUpdate =
+                            project.Root.Descendants(ns + PropertyName)
+                                .Where(p => p.Attribute("Version") != null && p.Attribute("Version")!.Value.Equals($"$({targetName})", StringComparison.OrdinalIgnoreCase));
+                        foreach (var packageReference in packageReferencesToUpdate)
+                        {
+                            if(packageReference.Attribute("Version") is not null)
+                            {
+                                packageReference.Attribute("Version")!.Value = targetValue;
+                            }
+                        }
```

Comment:
Similar to the comment above. You might need to set the type on `versionAttributesToUpdate` to be explicitly not null for it to resolve in the foreach properly. This avoids checking it is not null twice, since you already filter out the nulls in the Where.

```suggestion
                        var versionAttributesToUpdate =
                            project.Root.Descendants(ns + PropertyName)
                                .Select(p => p.Attribute("Version"))
                                .Where(va => va is not null && va.Value.Equals($"$({targetName})", StringComparison.OrdinalIgnoreCase));
                        foreach (var versionAttribute in versionAttributesToUpdate)
                        {
                            versionAttribute.Value = targetValue;
                        }
```